plugins/stylescombo/plugin.js

```diff
@@ -115,8 +139,10 @@
 							elements = elementPath.elements,
 							pathDepth = elements.length;
 
-						if (elementPath.elements[0].getName() === 'li') {
-							pathDepth = 2;
+						var cutOffIndex = getCutOffIndex(elements, ['ol', 'ul']);
```

Comment:
Just passing through, but I would have expected things like this to be accomplished by using a [dom.walker](http://docs.ckeditor.com/#!/api/CKEDITOR.dom.walker) to walk through the selection